libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -226,17 +251,28 @@ std::string YulUtilFunctions::revertWithError(
 {
 	solAssert((!_posVar.empty() && !_endVar.empty()) || (_posVar.empty() && _endVar.empty()));
 	bool const needsNewVariable = !_posVar.empty() && !_endVar.empty();
+	bool needsAllocation = true;
+
+	if (std::optional<size_t> size = staticEncodingSize(_parameterTypes))
+		if (ranges::all_of(_parameterTypes, [](auto const* type) { return type && type->isValueType(); }))
```

Comment:
We could assert that `type` is not `InaccessibleDynamic`, since that's actually a value type.